\documentclass[final]{beamer}
\usepackage{grffile}
\mode<presentation>{\usetheme{CambridgeUS}}

\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{graphicx}
\usepackage{array,booktabs,tabularx}
\usepackage{epstopdf}
\usepackage{colortbl, xcolor}
\newcolumntype{Z}{>{\centering\arraybackslash}X}

% rysunki
\usepackage{tikz}
\usepackage{ifthen}
\usepackage{xxcolor}
\usetikzlibrary{arrows}
\usetikzlibrary[topaths]
\usetikzlibrary{decorations.pathreplacing}
%\usepackage{times}\usefonttheme{professionalfonts}  % times is obsolete
\usefonttheme[onlymath]{serif}
\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                       % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

\newlength{\columnheight}
\setlength{\columnheight}{94cm}                 % Wysokość sekcji na bloki (Więcej miejsca na tytuł i autorów)
\renewcommand{\thetable}{}
\def\andname{,}
\authornote{}

\renewcommand{\APACrefatitle}[2]{}
\renewcommand{\bibliographytypesize}{\footnotesize} 
\renewcommand{\APACrefYearMonthDay}[3]{%
{\BBOP}{#1}
{\BBCP}
}

\begin{document}
\SweaveOpts{concordance=TRUE}

\date{}

\author{\large \underline{Jaros\l{}aw Chilimoniuk}\inst{1}*, Pawe\l{} Mackiewicz\inst{1} and Micha\l{} Burdukiewicz\inst{2}\\
\bigskip
\normaltext{*jaroslaw.chilimoniuk@gmail.com}}

\institute{\small{\textsuperscript{1}University of Wroc\l{}aw, Department of Genomics, Wroc\l{}aw, POLAND
\textsuperscript{2}Warsaw University of Technology, Warsaw, POLAND}}

\title{\huge Co-evolution of curli components CsgA and CsgB}

\begin{frame}
\begin{columns}
\begin{column}{.50\textwidth}                               % szerokość kolumny lewej
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}
\parbox[t][\columnheight]{\textwidth}
{

% \begin{block}{Nazwa}
% Tekst
% \end{block}
% \vfill

\begin{block}{Introduction}

<<knitrIntro, echo = FALSE, message=FALSE,warning=FALSE>>=

library(dplyr)
library(ggplot2)
library(reshape2)
@


CsgA, also known as a major curli component, is a secreted protein ubiquitous in biofilms of gram-negative bacteria. Thanks to its ability to create durable fibers, CsgA is a dominant proteinaceous scaffold of biofilms. In fact, CsgA belongs to amyloids, proteins that form fibers during a spontaneous aggregation. 

The presence of pre-formed amyloid fibers can accelerate aggregation of other amyloids. This process is known as cross-seeding. It is extremely sequence specific and can be restricted by a difference in a single amino acid. CsgA can be in vivo cross-seeded by its nucleator protein, CsgB, but also other CsgA fibrils.
\end{block}
\vfill


\begin{block}{Methods}
\begin{columns}

    \begin{column}{0.55\textwidth}
        \includegraphics[width=0.75\columnwidth]{schemat_psi_blast.eps}
    \end{column}
    \begin{column}{0.43\textwidth}
        We used CsgA from \textit{E. coli} K12 as a starting point of our search for CsgA homologs.
        
        \bigskip
        
        After five iterations of PSI-Blast, we found 5007 sequences producing significant alignments with E-value lower than the threshold.
        
        \bigskip
        
        We evaluated sequences using a simple heuristic approach to find the most probable candidates for the CsgA proteins.
        
        \bigskip
        
        Proteins were used to find appropriate genomes in the Nucleotide database.
        
        \bigskip
        
        Using genomic information, we reconstructed CsgBAC operons.
        
    \end{column}

\end{columns}

\end{block}
\vfill


\begin{block}{Results}


<<echo = FALSE,message=FALSE,fig.align='center',fig.width=15,fig.height=10,warning=FALSE>>=

plot(1:10)

@

% 
% \begin{columns}
%     \begin{column}{0.3\textwidth}
%         \includegraphics[width=0.8\columnwidth]{AG.eps}
%     \end{column}
%     \begin{column}{0.75\textwidth}
    
        \bigskip
        
        AmyloGram \citep{burdukiewicz_amyloidogenic_2017-1} was used to create reduced aa alphabet.
        The software assigned amino acids into 6 groups, by using eleven combinations of physicochemical properties:
        
        \bigskip 
        
        I) Lowest propensity to form $\beta$-sheets (Glycine)
        
        \bigskip
        
        II) The most hydrophilic, includes two strongly basic amino acids, highly flexible (Lysine, Proline, Arginine)
        
        \bigskip
        
        III) Strongly hydrophobic, highest propensity to form $\beta$-sheets (Isoleucyne, Leucine, Valine)
        
        \bigskip
        
        IV) Aromatic properties, the most hydrophilic, the least flexible, highest propensity to form $\beta$-sheets (Phenylalanine, Tryptophan, Tyrosine)
        
        \bigskip
        
        V) The least flexible (Alanine, Cysteine, Histidine, Methionine)
        
        \bigskip
        
        VI) Strongly hydrophilic and highly flexible (Aspartic Acid, Glutamic Acid, Asparagine, Glutamine, Serine, Threonine)
        
%     \end{column}
% \end{columns}

% AmyloGram was used to create reduced aa alphabet.
% The software assigned amino acids into 6 groups, by using eleven combinations of physicochemical properties.
% Amino acids where divided in subgroups:
% I) lowest propensity to form $\beta$-sheets
% II) the most hydrophilic, includes two strongly basic amino acids, highly flexible
% III) strongly hydrophobic, highest propensity to form $\beta$-sheets
% IV) aromatic properties, the most hydrophilic, the least flexible, highest propensity to form $\beta$-sheets
% V) the least flexible
% VI) strongly hydrophilic and highly flexible

\end{block}
\vfill
}
\end{minipage}
\end{beamercolorbox}
\end{column}


%new column ------------------------------------------------------    

\begin{column}{.51\textwidth}                                 % szerokość kolumny prawej
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}  
\parbox[t][\columnheight]{\textwidth}
{



\begin{block}{Conclusions}

The interplay of CsgA and CsgB suggests that if a mutation occurs in the region responsible for protein interaction, it should be compensated by mutations in other protein. We have not identified any simultaneous mutations between CsgA and CsgB. This may be due that single mutation in one region is not enough to change the protein function and to cause mutations in another protein. Probably, the compensation of single mutations by the regional structure of mentioned proteins is sufficient.

\end{block}
\vfill 

 % \begin{block}{Bibliography}
 %  \tiny{
 %  \bibliographystyle{apalike}
 %  \bibliography{tymczasowa}
 %  }
 %  \end{block}
  \vfill  
}
\end{minipage}
\end{beamercolorbox}
\end{column}
\end{columns}  
\end{frame}
\end{document}
